casts/modalNavigator/src/ModalNavigator/createModalNavigator.tsx

```diff
@@ -43,165 +33,44 @@ const ModalScreen = ({
     }, [name, descriptor.navigation]);
 
     return (
-        <NestedInModalContext.Provider value={hide}>
-            {descriptor.render()}
-        </NestedInModalContext.Provider>
+        <UIModalSheet visible={route.visible} onClose={hide} maxMobileWidth={maxMobileWidth}>
+            <NestedInModalContext.Provider value={hide}>
+                {descriptor.render()}
+            </NestedInModalContext.Provider>
+        </UIModalSheet>
     );
-};
-export class ModalController extends React.Component<ModalControllerProps> {
```

Comment:
Maybe to keep it as legacy until UIModalController is fully removed? Otherwise the migration process will turn into a nightmare...